test/Interop/Cxx/class/constructors-executable.swift

```diff
@@ -44,4 +44,18 @@ CxxConstructorTestSuite.test("TemplatedConstructor") {
   expectEqual(2, instance.value.i)
 }
 
+CxxConstructorTestSuite.test("implicit default ctor") {
+  // Make sure that fields of C++ structs are zeroed out.
+
+  let instance1 = ConstructorWithParam()
+  expectEqual(0, instance1.x)
+
+  let instance2 = IntWrapper()
+  expectEqual(0, instance2.x)
+
+  let instance3 = CopyAndMoveConstructor()
+  expectEqual(0, instance3.value)
```

Comment:
Sorry if I'm missing something obvious, but why doesn't `instance3.value` end up getting the `123` initial value from `Inputs/constructors.h`? Naïvely I would expect the compiler to respect any explicit initial values and simply zero out anything that was left uninitialized. (Or zero out everything before applying the explicit initial values, which amounts to the same thing.)

Reply:
Good point! This definitely deserves a comment.

The type `CopyAndMoveConstructor` is not default-initializable in C++, so the default initializer here is actually an implicitly generated deprecated Swift constructor that zeroes out all of the fields of the type. Using this constructor produces a deprecation warning that describes the issue, I added a typechecker test to verify that.